index-scheduler/src/lib.rs

```diff
@@ -604,6 +603,43 @@ mod tests {
         assert_eq!(tasks[2].status, Status::Enqueued);
     }
 
+    /// We send a lot of tasks but notify the tasks scheduler only once as
+    /// we send them very fast, we must make sure that they are all processed.
+    #[test]
+    fn process_tasks_inserted_without_new_signal() {
+        let (index_scheduler, handle) = IndexScheduler::test();
+
+        index_scheduler
+            .register(KindWithContent::IndexCreation {
+                index_uid: S("doggos"),
+                primary_key: None,
+            })
+            .unwrap();
+        index_scheduler
+            .register(KindWithContent::IndexCreation {
+                index_uid: S("cattos"),
+                primary_key: None,
+            })
+            .unwrap();
+        index_scheduler
+            .register(KindWithContent::IndexDeletion {
+                index_uid: S("doggos"),
+            })
+            .unwrap();
+
+        handle.wait_till(Breakpoint::Start);
+        handle.wait_till(Breakpoint::AfterProcessing);
+        handle.wait_till(Breakpoint::AfterProcessing);
+        handle.wait_till(Breakpoint::AfterProcessing);
```

Comment:
```suggestion
        handle.dont_block();
```

I'm not sure, but I think that's what you're trying to do?

Reply:
Not sure about that!